staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -301,6 +309,9 @@ func (le *LeaderElector) release() bool {
 		klog.Errorf("Failed to release lock: %v", err)
 		return false
 	}
+
+	le.observedRecordLock.Lock()
+	defer le.observedRecordLock.Unlock()
```

Comment:
Consider adding a setObservedRecord/getObservedRecord function to encapsulate this.

Reply:
I added: https://github.com/kubernetes/kubernetes/pull/97419/commits/2bd6b3970a33566d085c2b6d2947016b2762bb44